### PR TITLE
Mon-7153: bulk inserts index_data during config generation

### PR DESCRIPTION
**Pull request by garnier-quentin**
## Description

We have removed transaction and we use bulk inserts.

## Type of change

- [X] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 19.10.x
- [X] 20.04.x
- [X] 20.10.x
- [X] 21.04.x (master)

<h2> How this pull request can be tested ? </h2>

Nothing changes. It should work like before: add a service and get graphs in realtime.

## Checklist

- [X] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/class/config-generate/generate.class.php [hunk 1]
```diff
@@ -89,22 +89,34 @@ private function generateIndexData($localhost = 0)
...
-                    $stmt->execute();
+                    $bulkValues[] = '(' . 
+                        $this->backend_instance->db_cs->quote($host_id, PDO::PARAM_INT) . ',' . 
+                        $this->backend_instance->db_cs->quote($service_id, PDO::PARAM_INT) . ',' .
+                        $this->backend_instance->db_cs->quote(
+                            $host_instance->getString($host_id, 'host_name'),
+                            PDO::PARAM_STR
+                        ) . ',' . 
+                        $this->backend_instance->db_cs->quote(
+                            $service_instance->getString($service_id, 'service_description'),
```
Comment (sc979) [inline www/class/config-generate/generate.class.php:106]: kindly bind the values

File: www/class/config-generate/generate.class.php [hunk 2]
```diff
@@ -113,17 +125,35 @@ private function generateIndexData($localhost = 0)
...
-                    $stmt->bindValue(':host_name', '_Module_Meta', PDO::PARAM_STR);
-                    $stmt->bindValue(':service_description', 'meta_' . $meta_id, PDO::PARAM_STR);
-                    $stmt->bindParam(':host_id', $host_id, PDO::PARAM_INT);
-                    $stmt->bindParam(':service_id', $meta_service['service_id'], PDO::PARAM_INT);
-                    $stmt->execute();
+                    $bulkValues[] = '(' . 
+                        $this->backend_instance->db_cs->quote($host_id, PDO::PARAM_INT) . ',' . 
+                        $this->backend_instance->db_cs->quote($meta_service['service_id'], PDO::PARAM_INT) . ',' .
+                        $this->backend_instance->db_cs->quote('_Module_Meta', PDO::PARAM_STR) . ',' . 
+                        $this->backend_instance->db_cs->quote('meta_' . $meta_id, PDO::PARAM_STR) . ')';
```
Comment (sc979) [inline www/class/config-generate/generate.class.php:132]: kindly bind the values

File: www/class/config-generate/generate.class.php [hunk 3]
```diff
@@ -113,17 +125,35 @@ private function generateIndexData($localhost = 0)
...
+                            " ON DUPLICATE KEY UPDATE 
+                            host_name=VALUES(host_name), service_description=VALUES(service_description)"
+                        );
+                        $bulkCount = 0;
+                        $bulkValues = [];
+                    }
                 }
             }
 
-            $this->backend_instance->db_cs->commit();
```
Comment (sc979) [inline www/class/config-generate/generate.class.php:124]: to stay ISO, commit and rollback are required

**Review by sc979 — CHANGES_REQUESTED** [review]

File: www/class/config-generate/generate.class.php (hunk 3 above)
Comment (kduret) [reply www/class/config-generate/generate.class.php:124]: we don't need transaction here, it locks table index_data too much time, and it's not necessary

**Review by jeremyjaouen — APPROVED** [review]

**Review by sc979 — APPROVED** [review]

**Pull request merged**